Automattic/woocommerce-payments
# Allow users to continue onboarding if exited

Fixes #6931 
Requires https://github.com/Automattic/woocommerce-payments-server/pull/3830

#### Changes proposed in this Pull Request
- The PR uses the newly added `submittedDetails` account status parameter to modify the existing `UpdateBusinessDetailsTask` to prompt the user to finish onboarding.
- The new version of the tasks modifies the copy and action, using accounts_link instead of login_link. This allows to continue the onboarding in the step that was left, maintaining any prefilled information sent in the new onboarding flow.
- It introduces the `partially_onboarded` concept, where an account exists, it's connected, but haven't submitted all its details.









![image](https://github.com/Automattic/woocommerce-payments/assets/2612426/a8c2fa85-7a18-4feb-aad5-74cab68a5403)


#### Testing instructions

This should work in both New onboarding flow (with or without PO) and in the old flow.

[In new onboarding flow]
- After completing the hosted onboarding and being redirected to Stripe, click on return to site without adding your phone number.
- See the new card in the dashboard urging you to finish onboarding
- Click on “Complete registration”
- You are redirected to the “enter mobile phone” screen on Stripe
- Enter the test phone and see the data that you previously added in the hosted onboarding form here (meaning we are effectively recovering the same onboarding that was started)
- Note: clicking on return to site now after having been returned causes locally p1692334055012739-slack-C03KTTK2YMA (still not 100% sure that it’s expected :D so I’d appreciate confirming this in a non-docker environment too)
- After completing the onboarding the card in the dashboard is removed and everything gets back to normal

[In old onboarding flow]
- If leaving the onboarding on the “enter mobile phone” screen, you are redirected to Connect account. This is the existing behaviour and I think it’s the correct one because if there is no prefilled data and no phone number, so doens’t make sense to show the dashboard at this point, better to begin from zero.
- If you enter the phone number and also advance entering country and business structure, clicking on "Return to site" you are redirected to the Payments dashboard, and the card to continue onboarding is shown. At this point, when clicking, you won’t see prefilled data but you will continue in the step where you left it.

-------------------


**Post merge**



- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-6.4.0#allow-users-to-continue-onboarding-if-exited
